aten/src/ATen/core/NestedIntSymNodeImpl.h

```diff
@@ -179,8 +224,27 @@ class TORCH_API NestedIntSymNodeImpl : public SymNodeImpl {
 #undef DEFINE_NOT_SUPPORTED
 
  private:
+  // Why do we need to store val in additional to vec?
+  // It is not possible in eq to simply check whether the two vecs are the
+  // same because each nested int may be associated multiple vec, e.g. two
+  // NestedTensor with same raggedness but on different devices would have two
+  // different offsets tensors on different devices, but we'd still want their
+  // nested ints to compare equal, and we would do that by mapping both vec
+  // to the same val.
   int64_t val_;
   int64_t coeff_;
+  // Vec is a 1D tensor either representing offsets or lengths of a NestedTensor
+  // It should not be mutated
+  at::Tensor vec_;
+  // See NOTE [ NestedTensorVariant ]
+  NestedTensorVariant type_;
+  // Depending on the NestedTensorVariant we may want to either dispatch to
+  // Python or NestedTensor kernels. We dispatch to the Python kernel in the
+  // case the nested int corresponds to a NT python because we'd like for
+  // infra modes to be able to trace underneath.
+  c10::DispatchKeySet key_set_;
```

Comment:
dumb Q: do we need both a key set and NestedTensorVariant enum? doesn't one imply the other?

Reply:
Good point, we can probably get rid of explicitly storing the key set